controllers/agents/svc/serviceclass_controller.go

```diff
@@ -0,0 +1,240 @@
+		if err != nil {
+			return err
+		}
+		mappings[mapping.Name] = path
+	}
+
+	config, remote_namespace, err := r.getPrimazaKubeconfig(ctx, serviceClass.Namespace)
+	if err != nil {
+		return err
+	}
+	l.Info("remote cluster", "address", config.Host)
+
+	status := workercluster.TestConnection(ctx, config)
+	serviceClass.Status.Conditions = append(serviceClass.Status.Conditions, metav1.Condition{
+		Type:    "Connection",
+		Message: status.Message,
+		Reason:  string(status.Reason),
+		Status:  metav1.ConditionStatus(status.State),
+	})
+	if status.State == v1alpha1.ClusterEnvironmentStateOffline {
+		return fmt.Errorf("Failed to connect to cluster")
+	}
+
+	remote_client, err := client.New(config, client.Options{
+		Scheme: r.RemoteScheme,
+		Mapper: r.Mapper,
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, data := range services.Items {
+		sedMappings, err := LookupServiceEndpointDescriptor(mappings, data)
+		if err != nil {
+			l.Error(err, "Failed to lookup service endpoint descriptor values",
+				"name", data.GetName(),
+				"namespace", data.GetNamespace(),
+				"gvk", data.GroupVersionKind())
+		}
+
+		rs := v1alpha1.RegisteredService{
+			ObjectMeta: metav1.ObjectMeta{
+				// FIXME(sadlerap): this could cause naming conflicts; we need
+				// to take into account the type of resource somehow.
+				Name:      data.GetName(),
+				Namespace: remote_namespace,
+			},
+			Spec: v1alpha1.RegisteredServiceSpec{
+				ServiceEndpointDefinition: sedMappings,
+				ServiceClassIdentity:      serviceClass.Spec.ServiceClassIdentity,
+				HealthCheck:               serviceClass.Spec.HealthCheck,
+			},
+		}
+
+		if serviceClass.Spec.Constraints != nil {
+			rs.Spec.Constraints = &v1alpha1.RegisteredServiceConstraints{
+				Environments: serviceClass.Spec.Constraints.Environments,
+			}
+		}
+
+		if err := remote_client.Create(ctx, &rs); err != nil {
+			l.Error(err, "Failed to create registered service",
+				"service", data.GetName(),
+				"namespace", remote_namespace)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func LookupServiceEndpointDescriptor(mappings map[string]*jsonpath.JSONPath, service unstructured.Unstructured) ([]v1alpha1.ServiceEndpointDefinitionItem, error) {
+	var sedMappings []v1alpha1.ServiceEndpointDefinitionItem
+	for key, jsonPath := range mappings {
+		results, err := jsonPath.FindResults(service.Object)
+		if err != nil {
+			return nil, err
+		}
+		if len(results) == 1 && len(results[0]) == 1 {
+			value := fmt.Sprintf("%v", results[0][0])
+			sedMappings = append(sedMappings, v1alpha1.ServiceEndpointDefinitionItem{
+				Name:  key,
+				Value: value,
+			})
+		} else {
+			return nil, fmt.Errorf("jsonPath lookup into resource returned multiple results: %v", results)
+		}
+	}
+
+	return sedMappings, nil
+}
+
+const PRIMAZA_CONTROLLER_REFERENCE string = "primaza-kubeconfig"
+
+func (r *ServiceClassReconciler) getPrimazaKubeconfig(ctx context.Context, namespace string) (*rest.Config, string, error) {
```

Comment:
consider `pkg/primaza/clustercontext`

Reply:
This is subtly different from the relevant function in that package; we need to get both a kubeconfig and a namespace out from the secret.  Punt this for the future; added a todo so we don't forget.